replication-api/src/main/java/com/connexta/replication/api/data/Filter.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.data;
+
+import javax.annotation.Nullable;
+
+/** A filter for retrieving data from a specific site. */
+public interface Filter extends Persistable {
```

Comment:
Should filters also have a priority? When we are putting items in our priority queues I would imagine that the filter priority would be used as the item priority in the queue.

Reply:
Is that going to be used atm? I think we'll want to do it eventually but I'm not sure if it should be done right now. I could just add a priority that should be a number 1-10 if we know that's how we want to do it.

Reply:
I think it will be used when we switch over to queues in a couple sprints. We could wait till then but that's just one more thing to do and potentially a different version of the config.

Reply:
The 1-10 matches up with the queue broker api priority ranges so I think that would be ok. The other option would be to use the enums defined in the api but I think i like just a straight 1-10 better.

Reply:
I would prefer we be consistent and reuse the `PriorityLevel` enum by moving it into a common package to be shared by both the API and the queue.

If not then we should remove that enum and opt for a 1-10 everywhere.

Either way I opt for consistency.

Reply:
Agree we should try to be consistent. I think the 1-10 gives us more flexibility but I am ok either way. I do think we should include the priority in this iteration though.

Reply:
Let's go with a byte and fix it to 1-10. Let's make the change to the queue api to remove the enum and set it as a byte there as well. This should be a simple change that we could do at the same time for consistency.